react/features/subtitles/middleware.ts

```diff
@@ -134,18 +135,34 @@ function _endpointMessageReceived(store: IStore, next: Function, action: AnyActi
         name
     };
 
+    let newTranscriptMessage: ITranscriptMessage | undefined;
+
     if (json.type === JSON_TYPE_TRANSLATION_RESULT && json.language === language) {
         // Displays final results in the target language if translation is
         // enabled.
 
-        const newTranscriptMessage = {
+        newTranscriptMessage = {
             clearTimeOut: undefined,
-            final: json.text,
+            final: json.text?.trim(),
             participant
         };
 
-        _setClearerOnTranscriptMessage(dispatch, transcriptMessageID, newTranscriptMessage);
-        dispatch(updateTranscriptMessage(transcriptMessageID, newTranscriptMessage));
+        const cachedTranscriptMessage = state['features/subtitles']._cachedTranscriptMessages?.get(transcriptMessageID);
+
+        if (cachedTranscriptMessage) {
+            const cachedText = (cachedTranscriptMessage.stable || cachedTranscriptMessage.unstable)?.trim();
+            const newText = newTranscriptMessage.final;
+
+            if (cachedText && cachedText.length > 0 && newText && newText.length > 0
+                && newText.toLowerCase().startsWith(cachedText.toLowerCase())) {
```

Comment:
What are we doingh here? Removing the common part right? In that case a string slice will do, you don't need to complicate it with a regex.

Also, are intermediates translated too?

Reply:
> What are we doingh here? Removing the common part right? In that case a string slice will do, you don't need to complicate it with a regex.
Good idea! Thanks!

> Also, are intermediates translated too?

That I'm not sure. But it seems we never has been handling them and we have always set them to final.

Reply:
https://github.com/jitsi/jigasi/blob/2222c50c4010163521255c8db2a69c4e93c5f3ff/src/main/java/org/jitsi/jigasi/transcription/TranslationManager.java#L186
We only translate the final transcriptions, it was always like that.